emcc.py

```diff
@@ -2239,23 +2254,22 @@ def check_memory_setting(setting):
     if settings.ALLOW_MEMORY_GROWTH:
       max_mem = settings.MAXIMUM_MEMORY
 
-    shadow_size = max_mem // 8
-    settings.GLOBAL_BASE = shadow_size
-
-    sanitizer_mem = (shadow_size + webassembly.WASM_PAGE_SIZE) & ~webassembly.WASM_PAGE_SIZE
-    # sanitizers do at least 9 page allocs of a single page during startup.
-    sanitizer_mem += webassembly.WASM_PAGE_SIZE * 9
-    # we also allocate at least 11 "regions". Each region is kRegionSize (2 << 20) but
-    # MmapAlignedOrDieOnFatalError adds another 2 << 20 for alignment.
-    sanitizer_mem += (1 << 21) * 11
-    # When running in the threaded mode asan needs to allocate an array of kMaxNumberOfThreads
-    # (1 << 22) pointers.  See compiler-rt/lib/asan/asan_thread.cpp.
-    if settings.USE_PTHREADS:
-      sanitizer_mem += (1 << 22) * 4
+    # The total memory is 1/8th shadow and 7/8th usable memory
+    mem_size_including_shadow = max_mem * 8 / 7
```

Comment:
This adds `1/7`th more. Is that what you intended? The comment doesn't seem to match that as it speaks of `1/8`th sized pieces.

Reply:
I think this works out because `max_mem == mem_size_including_shadow * 7/8`.

Reply:
This is the tricky part of the calculation.. I'm working backward from  `user_mem = total_mem * 7 / 8` to get an expression for total_mem in term of user_mem.   Does my new comment explain things?

Reply:
Another way of putting it: if the shadow memory is 1/8 of the total then its 1/7th of the user memory.. right?

Reply:
Ah, now I see what you mean. Yes, the algebra checks out, here's how I'd write it:
```
total_mem = user_mem + shadow_mem
user_mem = 7 * shadow_mem

=>

shadow_mem = user_mem / 7

=>

total_mem = (8 / 7) * user_mem
```